src/status_im/contexts/wallet/events.cljs

```diff
@@ -80,6 +80,14 @@
                                    {:error %
                                     :event :wallet/get-accounts})}]]]}))
 
+(rf/reg-event-fx
+ :wallet/process-account-from-signal
+ (fn [{:keys [db]} [{:keys [address] :as account}]]
+   {:db (assoc-in db [:wallet :accounts address] (data-store/rpc->account account))
+    :fx [[:dispatch [:wallet/get-wallet-token-for-account address]]
+         [:dispatch [:wallet/request-new-collectibles-for-account-from-signal address]]
+         [:dispatch [:wallet/check-recent-history-for-account address]]]}))
```

Comment:
Introduced this new event to process accounts/addresses individually.

I have used the suffix `-from-signal` as right now we use this event for processing backed-up data but in the near future, we will handle sync messages (If an account is modified/created/deleted in a paired device, the other device is notified when online and this can handle those scenarios too)

Reply:
It's good to dispatch smaller events because they'll take less time to process, the problem is if they are touching the database too often and we have subscriptions mounted that react to it. We may end up having a lot of not needed rerenders.

To fix it we can write in the "subscribed" keys in batches, similar to what I did with collectibles.

Have you checked if this approach is triggering more rerenders than needed, for example, in the wallet home?

Reply:
I doubt touching app-db often as this event is dispatched only when we process the backed-up data* (during recovery) or receive any new data* from the paired device (yet to be done). 
*watch-only account or Key pair

In the case of recovery, The user won't immediately land in the wallet home after we receive the login signal (goes through Identifier, Notification, and get started page) which gives us a bit of time to update the app-db and unwrapping/update the assets data upon receiving from status-go before the user lands in the wallet tab. Later in the future, we will have a progress screen for backup recovery (just like Desktop).

Yes, there are re-renders whenever the loading state of an account changes and the card in the home updates. Apart from that, I don't see any unwanted re-renders.